packages/react/src/constants/device-sizes.constant.ts

```diff
@@ -41,40 +41,49 @@ const sizes = {
 export type Sizes = typeof sizes;
 
 export interface Breakpoints {
+  xsmall?: number;
   small?: number;
   medium?: number;
 }
 
-export const getSizesForBreakpoints = ({ small, medium }: Breakpoints) => {
+export const getSizesForBreakpoints = (breakpoints: Breakpoints) => {
   const newSizes = {
     ...sizes, // Note: this helps get the function from sizes
     ...fastClone(sizes), // Note: this helps to get a deep clone of fields like small, medium etc
   };
 
-  if (!small || !medium) {
+  if (!breakpoints) {
     return newSizes;
   }
 
-  const smallMin = Math.floor(small / 2);
-  newSizes.small = {
-    max: small,
-    min: smallMin,
-    default: smallMin + 1,
-  };
+  const { xsmall, small, medium } = breakpoints;
 
-  const mediumMin = newSizes.small.max + 1;
-  newSizes.medium = {
-    max: medium,
-    min: mediumMin,
-    default: mediumMin + 1,
-  };
+  if (xsmall) {
+    const xsmallMin = Math.floor(xsmall / 2);
+    newSizes.xsmall = {
+      max: xsmall,
+      min: xsmallMin,
+      default: xsmallMin + 1,
+    };
+  }
 
-  const largeMin = newSizes.medium.max + 1;
```

Comment:
check if this has to be added back

Reply:
It seems we are overriding the large min & default in this function. Also couldn't find anywhere where are using `newSizes.large`  in the code. I reverted the changes to keep changes minimal.